### PR TITLE
[backends] Add options for the param --category in help message for e…

### PR DESCRIPTION
**Pull request by harshalmittal4**
Addresses #511 

This PR adds the different available options for the param `--category` in the help message for each backend.

### CODE REVIEW

**Comment by inishchith** [comment]
@harshalmittal4 This looks to me as a nice addition. 
I was thinking of something like we could get for instance [CATEGORY](https://github.com/chaoss/grimoirelab-perceval/blob/70f82e64badf04e7de63b402107ba211e3be2ced/perceval/backends/core/github.py#L90) variable and update the `helptext` such as.

```python3
helptext = "type of the items to fetch ({categories})".format(categories=", ".join(GitHub.CATEGORIES))
```

This would avoid updating the `helptext` every time there's some change ( or update in categories of data retrieved ) in implementation in future.
Just a suggestion :)

@valeriocos what do you think?

**Comment by valeriocos** [comment]
Thank you @harshalmittal4 for the PR, the idea is good but the PR requires some refinements. I agree with the suggestion of @inishchith , which seems easier to maintain.

A possibility could be to change `setup_cmd_parser` from staticmethod to classmethod, and pass the categories value in this way:

```
class AskbotCommand(BackendCommand):
    """Class to run Askbot backend from the command line."""

    BACKEND = Askbot

    @classmethod
    def setup_cmd_parser(cls):
        """Returns the Askbot argument parser."""

        parser = BackendCommandArgumentParser(from_date=True,
                                              archive=True,
                                              categories=cls.BACKEND.CATEGORIES)

        # Required arguments
        parser.parser.add_argument('url',
                                   help="URL of the Askbot server")

        return parser
```

Then the `BackendCommandArgumentParser` should be changed to something like:
```
    def __init__(self, from_date=False, to_date=False, offset=False,
                 basic_auth=False, token_auth=False, archive=False,
                 aliases=None, categories=None):
        self._from_date = from_date
        self._to_date = to_date
        self._archive = archive

        self.aliases = aliases or {}
        self.parser = argparse.ArgumentParser()

        group = self.parser.add_argument_group('general arguments')
        group.add_argument('--category', dest='category',
                           help="type of the items to fetch (%s)" % ','.join(categories))
```

what do you think @sduenas ?

**Comment by harshalmittal4** [comment]
Actually, I was working on refining the PR(by modifying the tests) and trying to add some tests for the proposed change. 
@valeriocos, @inishchith, the idea is better and more structured definitely. But the thing that remains in this is that the param `category` is still be shown under the heading `general arguments` instead of `<backend> arguments` in the helptext. How can we correct this, if we go this way.

**Comment by harshalmittal4** [comment]
And yes, thanks @inishchith, really good having your suggestions :sweat_smile:

**Comment by harshalmittal4** [comment]
@valeriocos ping :)

**Comment by valeriocos** [comment]
@harshalmittal4 , I guess there is no problem in having the category param in the general arguments (since it's an argument valid for all backends). The backend params should be those ones that are backend-specific.

What do you think?

**Comment by harshalmittal4** [comment]
@valeriocos, as we are making the helptext for `--category` to be backend-specific (as it will include the different options that can be passed for param `--category` in that backend), so I initially thought to try including it in the group <backend> arguments.

**Comment by harshalmittal4** [comment]
@valeriocos, as we are making the helptext for --category to be backend-specific (as it will include the different options that can be passed for param --category in that backend), so I initially thought to try including it in the group backend arguments.

**Comment by harshalmittal4** [comment]
@valeriocos, as we are making the helptext for --category to be backend-specific (as it will include the different options that can be passed for param --category in that backend), so I initially felt iincluding it in the group backend arguments would be better :slightly_smiling_face:
But the above suggested way seems better with regards to implementation.
So shall I modify the PR as suggested ?

**Comment by valeriocos** [comment]
Yes please @harshalmittal4 . 
@jgbarah @sduenas any feedback on this PR?

**Comment by harshalmittal4** [comment]
@valeriocos, I modified the PR as required, 
@jgbarah, @sduenas, please see to this, thanks :)

File: perceval/backends/core/askbot.py [hunk 1]
```diff
@@ -526,12 +526,12 @@ class AskbotCommand(BackendCommand):
...
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Askbot argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/askbot.py]: please add a param per line, for instance:
```
parser = BackendCommandArgumentParser(from_date=True,
                                      archive=True, 
                                      categories=cls.BACKEND.CATEGORIES)
```

File: perceval/backends/core/bugzilla.py [hunk 2]
```diff
@@ -347,13 +347,13 @@ class BugzillaCommand(BackendCommand):
...
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Bugzilla argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               basic_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/bugzilla.py]: same comment as above, please add a param per line

File: perceval/backends/core/bugzillarest.py [hunk 3]
```diff
@@ -453,14 +453,14 @@ class BugzillaRESTCommand(BackendCommand):
...
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the BugzillaREST argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               basic_auth=True,
                                               token_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/bugzillarest.py]: same comment as above

File: perceval/backends/core/confluence.py [hunk 4]
```diff
@@ -282,12 +282,12 @@ class ConfluenceCommand(BackendCommand):
...
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Bugzilla argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/confluence.py]: same comment as above

File: perceval/backends/core/discourse.py [hunk 5]
```diff
@@ -399,13 +399,13 @@ class DiscourseCommand(BackendCommand):
...
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Discourse argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               token_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/discourse.py]: same comment as above

File: perceval/backends/core/dockerhub.py [hunk 6]
```diff
@@ -205,11 +205,11 @@ class DockerHubCommand(BackendCommand):
...
     BACKEND = DockerHub
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the DockerHub argument parser."""
 
-        parser = BackendCommandArgumentParser(archive=True)
+        parser = BackendCommandArgumentParser(archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/dockerhub.py]: same comment as above

File: perceval/backends/core/gerrit.py [hunk 7]
```diff
@@ -491,12 +491,12 @@ class GerritCommand(BackendCommand):
...
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Gerrit argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/gerrit.py]: same comment as above

File: perceval/backends/core/git.py [hunk 8]
```diff
@@ -341,11 +341,11 @@ def _pre_init(self):
...
         setattr(self.parsed_args, 'gitpath', git_path)
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Git argument parser."""
 
-        parser = BackendCommandArgumentParser(from_date=True, to_date=True)
+        parser = BackendCommandArgumentParser(from_date=True, to_date=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/git.py]: same comment as above

File: perceval/backends/core/github.py [hunk 9]
```diff
@@ -877,14 +877,14 @@ class GitHubCommand(BackendCommand):
...
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the GitHub argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               to_date=True,
                                               token_auth=False,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/github.py]: same comment as above

File: perceval/backends/core/gitlab.py [hunk 10]
```diff
@@ -628,13 +628,13 @@ class GitLabCommand(BackendCommand):
...
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the GitLab argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               token_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/gitlab.py]: same comment as above

File: perceval/backends/core/googlehits.py [hunk 11]
```diff
@@ -238,11 +238,11 @@ class GoogleHitsCommand(BackendCommand):
...
     BACKEND = GoogleHits
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the GoogleHits argument parser."""
 
-        parser = BackendCommandArgumentParser(archive=True)
+        parser = BackendCommandArgumentParser(archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/googlehits.py]: same comment as above

File: perceval/backends/core/groupsio.py [hunk 12]
```diff
@@ -267,12 +267,12 @@ def _pre_init(self):
...
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Groupsio argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
-                                              token_auth=True)
+                                              token_auth=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/groupsio.py]: same comment as above

File: perceval/backends/core/hyperkitty.py [hunk 13]
```diff
@@ -264,11 +264,11 @@ def _pre_init(self):
...
         setattr(self.parsed_args, 'dirpath', dirpath)
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the HyperKitty argument parser."""
 
-        parser = BackendCommandArgumentParser(from_date=True)
+        parser = BackendCommandArgumentParser(from_date=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/hyperkitty.py]: same comment as above

File: perceval/backends/core/jenkins.py [hunk 14]
```diff
@@ -246,11 +246,11 @@ class JenkinsCommand(BackendCommand):
...
     BACKEND = Jenkins
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Jenkins argument parser."""
 
-        parser = BackendCommandArgumentParser(archive=True)
+        parser = BackendCommandArgumentParser(archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/jenkins.py]: same comment as above

File: perceval/backends/core/jira.py [hunk 15]
```diff
@@ -409,13 +409,13 @@ class JiraCommand(BackendCommand):
...
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Jira argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               basic_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/jira.py]: same comment as above

File: perceval/backends/core/launchpad.py [hunk 16]
```diff
@@ -451,13 +451,13 @@ class LaunchpadCommand(BackendCommand):
...
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Launchpad argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               archive=True,
-                                              token_auth=False)
+                                              token_auth=False, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/launchpad.py]: same comment as above

File: perceval/backends/core/mattermost.py [hunk 17]
```diff
@@ -385,13 +385,13 @@ class MattermostCommand(BackendCommand):
...
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Meetup argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               token_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/mattermost.py]: same comment as above

File: perceval/backends/core/mbox.py [hunk 18]
```diff
@@ -328,11 +328,11 @@ class MBoxCommand(BackendCommand):
...
     BACKEND = MBox
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the MBox argument parser."""
 
-        parser = BackendCommandArgumentParser(from_date=True)
+        parser = BackendCommandArgumentParser(from_date=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/mbox.py]: same comment as above

File: perceval/backends/core/mediawiki.py [hunk 19]
```diff
@@ -528,12 +528,12 @@ class MediaWikiCommand(BackendCommand):
...
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the MediaWiki argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/mediawiki.py]: same comment as above

File: perceval/backends/core/meetup.py [hunk 20]
```diff
@@ -270,14 +270,14 @@ class MeetupCommand(BackendCommand):
...
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Meetup argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               to_date=True,
                                               token_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/meetup.py]: same comment as above

File: perceval/backends/core/nntp.py [hunk 21]
```diff
@@ -361,12 +361,12 @@ class NNTPCommand(BackendCommand):
...
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the NNTP argument parser."""
 
         parser = BackendCommandArgumentParser(offset=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/nntp.py]: same comment as above

File: perceval/backends/core/phabricator.py [hunk 22]
```diff
@@ -612,13 +612,13 @@ class PhabricatorCommand(BackendCommand):
...
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Phabricator argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               token_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/phabricator.py]: same comment as above

File: perceval/backends/core/pipermail.py [hunk 23]
```diff
@@ -144,11 +144,11 @@ def _pre_init(self):
...
         setattr(self.parsed_args, 'dirpath', dirpath)
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Pipermail argument parser."""
 
-        parser = BackendCommandArgumentParser(from_date=True)
+        parser = BackendCommandArgumentParser(from_date=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/pipermail.py]: same comment as above

File: perceval/backends/core/redmine.py [hunk 24]
```diff
@@ -287,13 +287,13 @@ class RedmineCommand(BackendCommand):
...
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Redmine argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               token_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/redmine.py]: same comment as above

File: perceval/backends/core/rss.py [hunk 25]
```diff
@@ -176,11 +176,11 @@ class RSSCommand(BackendCommand):
...
     BACKEND = RSS
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the RSS argument parser."""
 
-        parser = BackendCommandArgumentParser(archive=True)
+        parser = BackendCommandArgumentParser(archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/rss.py]: same comment as above

File: perceval/backends/core/slack.py [hunk 26]
```diff
@@ -431,13 +431,13 @@ class SlackCommand(BackendCommand):
...
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Slack argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               token_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/slack.py]: same comment as above

File: perceval/backends/core/stackexchange.py [hunk 27]
```diff
@@ -301,13 +301,13 @@ class StackExchangeCommand(BackendCommand):
...
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the StackExchange argument parser."""
 
         parser = BackendCommandArgumentParser(from_date=True,
                                               token_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/stackexchange.py]: same comment as above

File: perceval/backends/core/supybot.py [hunk 28]
```diff
@@ -264,15 +264,15 @@ class SupybotCommand(BackendCommand):
...
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Supybot argument parser."""
 
         aliases = {
             'dirpath': 'ircdir'
         }
         parser = BackendCommandArgumentParser(from_date=True,
-                                              aliases=aliases)
+                                              aliases=aliases, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/supybot.py]: same comment as above

File: perceval/backends/core/telegram.py [hunk 29]
```diff
@@ -269,7 +269,7 @@ def setup_cmd_parser():
         parser = BackendCommandArgumentParser(offset=True,
                                               token_auth=True,
                                               archive=True,
-                                              aliases=aliases)
+                                              aliases=aliases, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/telegram.py]: same comment as above

File: perceval/backends/core/twitter.py [hunk 30]
```diff
@@ -343,12 +343,12 @@ class TwitterCommand(BackendCommand):
...
 
-    @staticmethod
-    def setup_cmd_parser():
+    @classmethod
+    def setup_cmd_parser(cls):
         """Returns the Twitter argument parser."""
 
         parser = BackendCommandArgumentParser(token_auth=True,
-                                              archive=True)
+                                              archive=True, categories=cls.BACKEND.CATEGORIES)
```
Comment (valeriocos) [inline perceval/backends/core/twitter.py]: same comment as above

File: testing.json [hunk 31]
```diff
@@ -0,0 +1,9 @@
...om/orgs/Cloud-CV/events","hooks_url":"https://api.github.com/orgs/Cloud-CV/hooks","id":20895851,"issues_url":"https://api.github.com/orgs/Cloud-CV/issues","login":"Cloud-CV","members_url":"https://api.github.com/orgs/Cloud-CV/members{/member}","node_id":"MDEyOk9yZ2FuaXphdGlvbjIwODk1ODUx","public_members_url":"https://api.github.com/orgs/Cloud-CV/public_members{/member}","repos_url":"https://api.github.com/orgs/Cloud-CV/repos","url":"https://api.github.com/orgs/Cloud-CV"}],"organizations_url":"https://api.github.com/users/deshraj/orgs","public_gists":7,"public_repos":110,"received_events_url":"https://api.github.com/users/deshraj/received_events","repos_url":"https://api.github.com/users/deshraj/repos","site_admin":false,"starred_url":"https://api.github.com/users/deshraj/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/deshraj/subscriptions","type":"User","updated_at":"2019-04-12T14:28:52Z","url":"https://api.github.com/users/deshraj"}},"origin":"https://github.com/Cloud-CV/EvalAI","perceval_version":"0.12.10","tag":"https://github.com/Cloud-CV/EvalAI","timestamp":1555132034.144707,"updated_on":1477019450.0,"uuid":"e490cdcd288c5e3c2ed4fe4100042b4cc33e1932"}
```
Comment (valeriocos) [inline testing.json]: why this file is needed?

File: perceval/backend.py [hunk 32]
```diff
@@ -291,7 +291,7 @@ class BackendCommandArgumentParser:
     """
     def __init__(self, from_date=False, to_date=False, offset=False,
                  basic_auth=False, token_auth=False, archive=False,
-                 aliases=None):
+                 aliases=None, categories=[]):
         self._from_date = from_date
         self._to_date = to_date
         self._archive = archive
```
Comment (valeriocos) [inline perceval/backend.py]: I would store the categories in a variable to simplify the tests, something like:
`self._categories = categories`

what do you think ?

**Comment by valeriocos** [comment]
please, @jgbarah @sduenas provide feedback on this PR when you have time.

File: perceval/backend.py (hunk 32 above)
Comment (harshalmittal4) [reply perceval/backend.py]: Agreed @valeriocos, its more organised, I will update, thanks :)

File: testing.json (hunk 31 above)
Comment (harshalmittal4) [reply testing.json]: Sorry, it was for some other checking, by  mistake.

File: perceval/backends/core/askbot.py (hunk 1 above)
Comment (sduenas) [inline perceval/backends/core/askbot.py]: I would put `categories` in a new line to align it with the style followed for the rest of the parameters. Moreover, if we change the definition of `setup_cmd_parser`, it should be also modified in the parent class.

File: testing.json (hunk 31 above)
Comment (harshalmittal4) [reply testing.json]: Sorry my mistake @valeriocos, it was for some other checking, removed.

File: perceval/backends/core/askbot.py (hunk 1 above)
Comment (harshalmittal4) [reply perceval/backends/core/askbot.py]: Done @sduenas!

File: perceval/backend.py [hunk 33]
```diff
@@ -291,7 +291,7 @@ class BackendCommandArgumentParser:
     """
     def __init__(self, from_date=False, to_date=False, offset=False,
                  basic_auth=False, token_auth=False, archive=False,
-                 aliases=None):
+                 aliases=None, categories=[]):
```
[the remark this answers is not in the document]
Comment (harshalmittal4) [reply perceval/backend.py]: Done, please review.

**Comment by harshalmittal4** [comment]
Hey @sduenas @valeriocos
I have updated and added the tests assuming `categories` to be a required argument, want some guidance on this PR for changes required if any, please review..
(The commits squashed and rebased with master)

File: tests/test_backend.py [hunk 34]
```diff
@@ -539,15 +541,21 @@ class TestBackendCommandArgumentParser(unittest.TestCase):
     def test_argument_parser(self):
         """Test if an argument parser object is created on initialization"""
 
-        parser = BackendCommandArgumentParser()
+        parser = BackendCommandArgumentParser(categories=MockedBackendCommand.BACKEND.CATEGORIES)
         self.assertIsInstance(parser.parser, argparse.ArgumentParser)
 
+    def test_categories(self):
+        """Test whether _categories is initialised"""
```
Comment (valeriocos) [inline tests/test_backend.py]: Please, add an empty line between the comment and the code. In addition, I guess initialised should be initialized (a quick search on the perceval repo shows that `initialized` is the preferred option, https://github.com/chaoss/grimoirelab-perceval/search?q=initialized&unscoped_q=initialized)

Comment (harshalmittal4) [reply tests/test_backend.py]: Done, thanks :)

**Comment by harshalmittal4** [comment]
@valeriocos, @sduenas please review when you have time, thanks :)

**Comment by valeriocos** [comment]
Thanks @harshalmittal4 , LGTM. What do you think @sduenas  ?

**Comment by harshalmittal4** [comment]
I also felt this earlier once, @sduenas, updated :)

**Pull request merged**